packages/nx/src/tasks-runner/utils.ts

```diff
@@ -253,24 +254,16 @@ function assertOutputsAreValidType(outputs: unknown) {
 export function validateOutputs(outputs: string[]) {
   assertOutputsAreValidType(outputs);
 
-  const invalidOutputs = new Set<string>();
-
-  for (const output of outputs) {
-    if (!/^!?{[\s\S]+}/.test(output)) {
-      invalidOutputs.add(output);
-    }
-  }
-  if (invalidOutputs.size > 0) {
-    throw new InvalidOutputsError(outputs, invalidOutputs);
-  }
+  nativeValidateOutputs(outputs);
 }
 
 export function transformLegacyOutputs(
   projectRoot: string,
-  error: InvalidOutputsError
+  outputs: string[],
+  transformableOutputs: Set<string>
```

Comment:
What if we determine what is transformable here in this function rather than passing it in?